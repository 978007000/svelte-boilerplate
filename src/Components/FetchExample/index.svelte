<script>
  import { onMount } from "svelte";
  import { Http } from "@Services/Http.js";

  let data = [];

  onMount(async function() {
    console.log(process.env.API_URL);
    const response = await Http.Request('GET', '/link/');
    data = response;
  });
</script>

<style src="./style.scss">

</style>

<ul>
  {#each data as link}
    <li>
      <a href={link.url}>{link.title}</a>
    </li>
  {/each}
</ul>
