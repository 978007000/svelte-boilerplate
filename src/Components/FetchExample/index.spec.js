import FetchExample from './index.svelte';

describe('Sample', () => {
	it('is empty by default', (next) => {
		const target = document.createElement('div');
		const sample = new FetchExample({ target });
        setTimeout(() => {
			const { firstElementChild: element } = target;
			expect(element.children.length).toBe(2);

            next();
		}, 10);
	});
});