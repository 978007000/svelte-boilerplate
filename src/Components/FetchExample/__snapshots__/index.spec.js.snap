// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample is empty by default 1`] = `
<body>
  <div>
    <ul>
      <li>
        <a
          href="test"
        >
          test
        </a>
         
      </li>
      <li>
        <a
          href="test2"
        >
          test2
        </a>
         
      </li>
    </ul>
  </div>
</body>
`;
